ts/components/LabelledItem.tsx

```diff
@@ -74,144 +70,140 @@ type Props = WithTestID<CommonProp> &
       }>
   );
 
-export class LabelledItem extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = { isEmpty: true, hasFocus: false };
-  }
+// eslint-disable-next-line sonarjs/cognitive-complexity
+export const LabelledItem: React.FC<Props> = (props: Props) => {
+  const [isEmpty, setIsEmpty] = useState(true);
+  const [hasFocus, setHasFocus] = useState(false);
 
   /**
    * check if the input is empty and set the value in the state
    */
-  private checkInputIsEmpty = (text?: string) => {
-    const isEmpty = text === undefined || text.length === 0;
-    if (isEmpty !== this.state.isEmpty) {
-      this.setState({ isEmpty });
-    }
+  const checkInputIsEmpty = (text?: string) => {
+    const isInputEmpty = text === undefined || text.length === 0;
+    setIsEmpty(isInputEmpty);
   };
 
   /**
    * handle input on change text
    */
-  private handleOnChangeText = (text: string) => {
-    if (this.props.type === "text" && this.props.inputProps.onChangeText) {
-      this.props.inputProps.onChangeText(text);
+  const handleOnChangeText = (text: string) => {
+    if (props.type === "text" && props.inputProps.onChangeText) {
+      props.inputProps.onChangeText(text);
     }
-    this.checkInputIsEmpty(text);
+    checkInputIsEmpty(text);
   };
 
   /**
    * handle masked input on change text
    */
-  private handleOnMaskedChangeText = (formatted: string, text?: string) => {
-    if (
-      this.props.type === "masked" &&
-      this.props.inputMaskProps.onChangeText
-    ) {
-      this.props.inputMaskProps.onChangeText(formatted, text);
+  const handleOnMaskedChangeText = (formatted: string, text?: string) => {
+    if (props.type === "masked" && props.inputMaskProps.onChangeText) {
+      props.inputMaskProps.onChangeText(formatted, text);
```

Comment:
```suggestion
    if (props.type === "text") {
      props.inputProps.onChangeText?.(text);
    }
```
Could it be a code more readable?

Reply:
@Undermaken I'm not sure to have understand what do you mean.
The check is needed for type inference purpose and `props.inputProps.onChangeText?.(text);` is different from `props.inputMaskProps.onChangeText(formatted, text);`

Reply:
Sorry it was a bad suggestion (I updated it)
The new code is about to avoid the undefined check using the optional chaining

Reply:
Oh yes, sure! Updated in [dac0bc1](https://github.com/pagopa/io-app/pull/3020/commits/dac0bc133c52a0ccb2e69f05b2a33fdef4bebf5c)